File: NextDoorITFrontend/src/app/models/service.model.ts
import { ReviewRating } from "./reviewRating.model";
import { ServiceCategory } from "./serviceCategory.model";

export class Service {
  id?: string;
  serviceName?: string;
  description?: string;
  category?: any = ServiceCategory;
  userOverallRating?: number;
  reviewRatings?: any = ReviewRating;
  imageId?: string;
  duration?: string;
  price?: string;
}

File: NextDoorITBackend/src/main/java/com/inn/nextDoorIt/entity/ServiceModel.java
package com.inn.nextDoorIt.entity;

import com.inn.nextDoorIt.entity.Category;
import lombok.Data;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.DynamicUpdate;

import javax.persistence.*;

@Data
@Entity
@DynamicUpdate
@DynamicInsert
@Table(name = "services")

public class ServiceModel {
    private static final long serialVersionUID = 1L;


    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    @Column(name = "id")
    private Integer id;
    private String serviceName;
    private String description;
    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "category_id")
    private Category category;
    private String imageId;
    private long price;
    private long duration;
}
